examples/basic_diesel/src/main.rs

```diff
@@ -0,0 +1,168 @@
+extern crate futures;
+extern crate gotham;
+extern crate hyper;
+extern crate mime;
+extern crate gotham_middleware_diesel;
+extern crate diesel;
+extern crate r2d2_diesel;
+extern crate r2d2;
+extern crate basic_diesel;
+extern crate serde_json;
+
+use hyper::{Response, StatusCode};
+use gotham::state::{State, FromState};
+use gotham::router::Router;
+use gotham::pipeline::new_pipeline;
+use gotham::router::builder::*;
+use gotham::router::route::dispatch::{new_pipeline_set, finalize_pipeline_set};
+use gotham::handler::HandlerFuture;
+use gotham::http::response::create_response;
+use gotham_middleware_diesel::DieselMiddleware;
+use gotham::handler::IntoHandlerError;
+use diesel::sqlite::SqliteConnection;
+use r2d2_diesel::ConnectionManager;
+use r2d2::{Pool, PooledConnection};
+use futures::{future, Future, Stream};
+use std::str;
+use basic_diesel::models::NewPost;
+
+
+// The URL of the database.
+static DATABASE_URL: &'static str = "posts.db";
```

Comment:
I note the database was committed, suggest removing the binary, it won't be necessary for folks.